spec/unit/resource/windows_feature_powershell_spec.rb

```diff
@@ -54,4 +50,14 @@
     resource.feature_name "SNMP"
     expect(resource.feature_name).to eql(["snmp"])
   end
+
+  it "install a single feature" do
+    resource.feature_name "snmp"
+    expect { resource.action :install }.not_to raise_error
+  end
+
+  it "install multi feature" do
+    resource.feature_name "SNMP, DHCP"
+    expect { resource.action :install }.not_to raise_error
+  end
```

Comment:
These tests don't test the behavior that was added in this PR.

What do you think about some tests like:

```
it "does not attempt to install features that have been removed" do
  node["powershell_features_cache"]["disabled"] = ["dhcp"]
  node["powershell_features_cache"]["removed"] = ["snmp"]
  resource.feature_name "dhcp, "snmp"

  expect(resource.features_to_install).to eq(["dhcp"])
end

it "attempts to install features that have been removed when source is set" do
  node["powershell_features_cache"]["disabled"] = ["dhcp"]
  node["powershell_features_cache"]["removed"] = ["snmp"]
  resource.feature_name "dhcp, "snmp"
  resource.source "some appropriate value"

  expect(resource.features_to_install).to eq(["dhcp", "snmp"])
end
```

Reply:
In process to add these tests, Was getting `undefined method `[]=' for nil:NilClass` while implementing these.

Reply:
Depending on what the error was, you might need to add something like `node["powershell_features_cache"] ||= {}`.

Reply:
Resolved now and updated, Thank you @phiggins !!